VEBT-2122, VEBT-3655, VEBT-3656, VEBT-3657, VEBT-3658, VEBT-3659, VEBT-3660: Form 0976 step 1

thomaskrenzke opened the pull request:
## Are you removing, renaming or moving a folder in this PR?
- [X] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [X] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

This PR implements the pages needed for Step 1 of the 0976 form. It's a lot of code, but nothing too fancy or out-of-the-ordinary. Everything is using standard platform-supplied components and the usual form config parameters. The 'acknowledgement' pages are basically identical except for the text. For the 'validating initials match the users name' in the acknowledgements pages, the logic from the 0839 was borrowed.

## Related issue(s)

[figma](https://www.figma.com/design/2L0vCxmiTxz5r6i0QAhErY/VEBT-Form-22-0976--Approval-of-a-Program-in-a-Foreign-Country?node-id=693-54082&t=kiP8UcnRqbsjbN1d-0)
[VEBT-2122](https://jira.devops.va.gov/browse/VEBT-2122)
[VEBT-3655](https://jira.devops.va.gov/browse/VEBT-3655)
[VEBT-3656](https://jira.devops.va.gov/browse/VEBT-3656)
[VEBT-3657](https://jira.devops.va.gov/browse/VEBT-3657)
[VEBT-3658](https://jira.devops.va.gov/browse/VEBT-3658)
[VEBT-3659](https://jira.devops.va.gov/browse/VEBT-3659)
[VEBT-3660](https://jira.devops.va.gov/browse/VEBT-3660)

## Testing done

Unit tests written, manual testing done locally, formal QA to follow when this is on staging.

## Screenshots
<img width="1314" height="2030" alt="name" src="https://github.com/user-attachments/assets/4be28e15-d6db-45fb-bc4d-f6c63b38f231" />

<img width="1314" height="1877" alt="what_to_expect" src="https://github.com/user-attachments/assets/6282dcaa-05c7-4010-a4f5-5efb25b4ff4a" />

<img width="1240" height="2294" alt="acknowledgement_1" src="https://github.com/user-attachments/assets/18b7bf9d-cff8-4e7b-8a5f-b3ee438595c1" />

<img width="1240" height="2327" alt="acknowledgement_1_error" src="https://github.com/user-attachments/assets/a1c0bf7e-5414-4cdf-9eaa-e174d91ec7a1" />

<img width="1240" height="2531" alt="acknowledgement_2" src="https://github.com/user-attachments/assets/f16b213c-98ae-4d81-9885-bf096f97fd49" />


<img width="1240" height="2573" alt="acknowledgement_3" src="https://github.com/user-attachments/assets/e5d3d503-15be-40f7-873f-c87e347ff10f" />

<img width="1314" height="2347" alt="acknowledgement_4" src="https://github.com/user-attachments/assets/56e57300-abab-47c4-949d-5b7913667c99" />

<img width="1240" height="2014" alt="acknowledgement_5" src="https://github.com/user-attachments/assets/b1e341b6-2120-4c16-8b52-5b5feecea530" />

## What areas of the site does it impact

*(Describe what parts of the site are impacted **if** code touched other areas)*

## Acceptance criteria

### Quality Assurance & Testing

<img width="1848" height="417" alt="0976_coverage" src="https://github.com/user-attachments/assets/988ce0af-dace-4c19-bebd-b75901d48bec" />


- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

thomaskrenzke committed:
0976 page 1 step 1

thomaskrenzke committed:
update need help component of 0976

thomaskrenzke committed:
0976 what to expect page

thomaskrenzke committed:
form 0976 first acknowledgement page

thomaskrenzke committed:
a couple more acknowledgement pages for 0976

thomaskrenzke committed:
fourth acknowledgement page for 0976

thomaskrenzke committed:
fifth acknowledgement page for 0976

thomaskrenzke committed:
use custom component for initials input to make sure its uppercase

thomaskrenzke committed:
refactor acknowledgement 1 to use same components as other steps

thomaskrenzke committed:
make sure link is external within form page

GovCIOLiz reviewed (changes requested):
- Add `useCustomScrollAndFocus: true` to this form as well in config so the focus goes to the page h3s as expected
- We have an A11y issue with the initials inputs not having a label to read for screen readers, and so at present - litany of text is being read out
- Please add in a screenshot showing test coverage so we can make sure we are staying above 80% on forms

thomaskrenzke committed:
add custom scroll to 0976 form config

thomaskrenzke committed:
add helpers unit spec to increase coverage

thomaskrenzke committed:
add ariaDescribedBy to initials inputs, remove "please" from error messages

thomaskrenzke commented:
@GovCIOLiz I've added the `useCustomScrollAndFocus` flag, as well as some test coverage where feasible. The one low spot in the testing is with the `InitialsInput` component that I made. It's just a wrapper around a `<VaTextInput/>` which capitalizes all the input. But since `<VaTextInput/>` is a custom component and has a shadow dom I haven't found a good way to test sending input events to it and exercising the custom `onInput` handler I'm using. Let me know if you have any ideas on this.

For the a11y stuff, I added an `ariaDescribedBy` field to each of the initials inputs. It is read by the screen reader, but the long title sentence is still read as well. Not sure if that's an improvement or not.

GovCIOLiz commented:
@thomaskrenzke For the acknolwedgement pages, you included the verbiage for those pages within the input configuration for the `textUI`. We instead, need to pull out the verbiage and put it into `descriptionUI`s between the `titleUI` and `textUI` config for each page. That is why the screenreader is reading out everything. Additionally, design needs to give us a label name for the initials inputs - I would say "Enter your initials", but we'll ask.

You should have something like this as example:
```  uiSchema: {
    ...titleUI('Institution Acknowledgements (1 of 5)'),
    ...descriptionUI(description),
    acknowledgement7: {
      'ui:title': 'Enter your initials',
      'ui:webComponentField': InitialsInput,
      'ui:options': {```

thomaskrenzke committed:
move large text blob out of initials input, add descriptive label
